src/main/scala/rps/models/Move.scala

```diff
@@ -0,0 +1,52 @@
+package rps.models
+
+sealed trait Move
+object Move {
+  case object Rock extends Move // 🪨
+  case object Paper extends Move // 📄
+  case object Scissors extends Move // ✂️
+  val moves = List(Rock, Paper, Scissors)
+
+  /**
+    * Parses `String` into `Move`
+    *
+    * @param input
+    * @return
+    */
```

Comment:
Non so quanto siano utili questi commenti.

Noi in generale tendiamo a usare pochi commenti, solo quando effettivamente aggiungono informazioni soprendenti o difficili da notare (o spiegano perché è stata fatta una scelta). Quindi in questo caso ometteremmo questo commento e quello su `encode` perché aggiungono molto poco rispetto ai nomi e ai tipi delle funzioni.

Forse terremmo solo
```scala
// Generates a user-friendly representation of Move
```
sull'ultima, o neanche quella se cambiamo il nome con "prettyPrint" o qualcos'altro del genere più autoesplicativo.

La logica è evitare commenti quando si può chiarire il concetto dal codice, visto che poi vanno mantenuti in sync e comunque occupano spazio. A volte però la spingiamo troppo in là e finiamo per avere cose meno comprensibili per la mancanza di commenti, temo.

Questo per dire: per me OK lasciare o togliere, ma non vedrai la pratica di commentare tutti i metodi pubblici nel nostro codice, cerchiamo di puntare più sui nomi.

Reply:
Sono d'accordo col tuo punto di vista. È una cosa che mi porto dietro da un progetto dove avevamo dei dev a cui piaceva che questi commenti fosser visibili nel loro IDE sull'hover di una variabile. Ciò non toglie che il codice migliore è quello che non devi scrivere =)